torchtext/experimental/transforms.py

```diff
@@ -174,6 +175,143 @@ def to_ivalue(self):
         return TextSequentialTransforms(OrderedDict(module_list))
 
 
+_pretrained_spm = ['text_unigram_15000', 'text_unigram_25000', 'text_unigram_50000',
+                   'text_bpe_15000', 'text_bpe_25000', 'text_bpe_50000']
+
+
+def pretrained_spm(spm_model='text_unigram_25000'):
+    r"""Generate a pretrained sentencepiece model.
+        The model was trained with torchtext.datasets.WikiText103, torchtext.datasets.EnWik9 and BookCorpus.
+        Both BPE and unigram methods were used to train the model (for more details please refer to
+        SentencePiece GitHub https://github.com/google/sentencepiece). We also provide the pretrained model
+        with a different size of the vocabulary (i.e. 15000, 25000, 50000).
+
+    Args:
+       spm_model: the pretrained sentencepiece model names. Default: 'text_unigram_25000'. The following pretrained sentencepiece models are provided:
+            - text_unigram_15000
+            - text_unigram_25000
+            - text_unigram_50000
+            - text_bpe_15000
+            - text_bpe_25000
+            - text_bpe_50000
+            Otherwise, the file path to the user-provided sentencepiece model is required.
+
+    Examples:
+        >>> from torchtext.experimental.transforms import pretrained_spm
+        >>> sp_model = pretrained_spm('text_unigram_25000')
+
+    """
+    if spm_model in _pretrained_spm:
+        spm_model = download_from_url('https://pytorch.s3.amazonaws.com/models/text/pretrained_spm/{}.model'.format(spm_model))
+        return load_sp_model(spm_model)
+    else:
+        raise RuntimeError('The pretrained sentencepiece model is not valid')
+
+
+def sentencepiece_tokenizer(spm_model):
+    r"""factory function to generate SentencePieceTokenizer with pretrained sp model
+
+    Args:
+       spm_model: the sentencepiece model instance
+
+    Examples:
+        >>> import torch
+        >>> from torchtext.experimental.transforms import SentencePieceTokenizer
+        >>> from torchtext.experimental.transforms import pretrained_spm
+        >>> sp_model = pretrained_spm('text_unigram_25000')
+        >>> spm_tokenizer = sentencepiece_tokenizer(sp_model)
+        >>> jit_spm_tokenizer = torch.jit.script(spm_tokenizer)
+    """
+    return SentencePieceTokenizer(spm_model)
```

Comment:
can you explain the reasoning behind this function? why not just have the class?

Reply:
As we discussed above, users will call this factory func, instead of the original class (a.k.a. `SentencePieceTokenizer`). We will have more flexibility to add new arguments without worry about BC breaking.